content/articles/working-with-csv-files-in-python/index.md

```diff
@@ -0,0 +1,253 @@
+
+- **Using CSV.DictReader() class:** The CSV file is opened using open(), then read using the CSV module's DictReader class, which works like a reader but converts CSV data to a dictionary. The file's first line contains dictionary keys.
+
+- **Using pandas.read_csv() method:** Using pandas library methods to read a CSV file is straightforward. Pandas read csv() reads data from CSV files.
+
+Consider the `fonteds.csv` CSV file. This is the file used to illustrate one of the methods.
+
+![Giant file](engineering-education/working-with-csv-files-in-python/image4.png
+)
+
+```python 
+import pandas
+csvFile = pandas.read_csv('fonteds.csv')
+print(csvFile)
+```
+
+The output 
+
+```bash 
+SCHOOL            CEO         YEAR
+
+0 KU    JACOB MUDAVAI         2010
+
+1 MUST   GARETH JASON         2015
+
+2 MMU   DICKSON NJOGU         2013
+```
+
+From the code above, `import pandas` is used to import the pandas module, `csvFile = pandas.read_csv('fonteds.csv')` is used to read the fonted.csv file and `print(csvFile)` is used to output the read csv file. 
+
+### Working with large CSV files in Python
+When dealing with CSV data, normally read it in using pandas before munging and analyzing it. Due to memory constraints, reading huge files straight into pandas may be difficult (or impossible) on a consumer machine.
+
+While it is simple to load data from CSV files into a database, there may be situations when you don't have access to or don't want to set up a database server. However, if you just need to look at data in these big files for a short time, here is one method to accomplish it using Python, and pandas
+
+Here is a method for handling large.csv files. The dataset we will be using is [gender_voice_dataset](/engineering-education/working-with-csv-files-in-python/voice.csv).
+
+#### Using pandas.read_csv
+Large files may be handled by reading them in pieces of manageable size, processing them before reading the next part. The chunk size option determines the number of lines. This method returns an iterator. For processing, a little file is read at a time.
+
+To read a dataset without chunks, use the code below:
+
+```python
+
+import pandas as pd # import pandas module
+import numpy as np  # Import numpy module
+import time         # import time module
+
+s_time = time.time() # This initilizes time module
+df = pd.read_csv("voice.csv") # This captures the time taken to read data from our file 
+e_time = time.time()
+
+print("Read without chunks: ", (e_time-s_time), "seconds") # print command is used to output the line specified while e_time-s_time outputs the time in seconds
+
+df.sample(10) # This specifies the time taken, 10 seconds
+
+```
+
+![Output](/engineering-education/working-with-csv-files-in-python/image1.png)
+
+
+### Convert multiple JSON files to CSV files
+A JSON file contains basic data structures and objects in JavaScript Object Notation (JSON). Sending data between an internet app and a server is the most common use case.

```

Comment:
Change "Sending data between an internet app and a server is the most common use case." to "The most common use case is sending data between an internet app and a server."